app/src/main/java/com/github/jorgecastillo/kotlinandroid/data/HeroesRepository.kt

```diff
@@ -2,7 +2,11 @@ package com.github.jorgecastillo.kotlinandroid.data
 
 import com.github.jorgecastillo.kotlinandroid.data.datasource.remote.fetchAllHeroes
 import com.github.jorgecastillo.kotlinandroid.data.datasource.remote.fetchHeroesFromAvengerComics
+import com.karumi.marvelapiclient.model.CharacterDto
+import kategory.HK
 
-fun getHeroesWithCachePolicy() = fetchAllHeroes()
+inline fun <reified F> getHeroesWithCachePolicy(): HK<F, List<CharacterDto>> =
```

Comment:
`F` needs to be reified to look up the registered instances. Kotlin forces you to use inline functions when reifying type args. The tagless style even in scala `carries` the entire program paramtrization so it can be made concrete at the end.
In scala:
```scala
def doSomething[F[_] : Monad] = Monad[F].flatMap...
```
In Kotlin
```kotlin
inline fun <F> doSomething(MF: Monad<F> = monad()) = M.flatMap(...
```

This has always been the same in kategory and not sure there is another way to do it in Kotlin. What do you mean by?: 

> I feel like the usage of implicits was much more transparent to the user before the later changes on the global instances system.